File: taskapi/src/main/kotlin/com/example/taskapi/model/Task.kt
package com.example.taskapi.model

import jakarta.persistence.*

@Entity
@Table(name = "tasks")
data class Task(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    var title: String,

    @Column(nullable = true)
    var description: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(nullable =  false)
    var priority: TaskPriority,

    @ManyToOne(optional = false)
    @JoinColumn(name = "board_id")
    val board: Board,

    @ManyToOne
    @JoinColumn(name = "column_id")
    val column: BoardColumn,

    @ManyToOne
    @JoinColumn(name = "assignee_id")
    var assignee: User? = null,

    @OneToMany(mappedBy = "task", cascade = [CascadeType.ALL], orphanRemoval = true)
    var comments: List<TaskComment> = emptyList(),

    @Embedded
    var audit: AuditFields = AuditFields()
): BaseEntity()

File: task-management-frontend/src/types/task.ts
import type { UserResponse } from './user'

export interface AssignTaskRequest {
  assigneeId: number
}

export interface CreateTaskRequest {
  title: string
  boardId: number
  description?: string
  status: string
  priority: string
  assigneeId?: number
}

export interface TaskResponse {
  id: number
  boardId: number
  columnId: number
  columnName: string
  title: string
  description: string
  assignee?: UserResponse
  priority: string
  createdAt: string
  updatedAt: string
}

export interface UpdateTaskRequest {
  title?: string
  description?: string
  status?: string
  priority?: string
  assigneeId?: number
}
